src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -684,208 +684,213 @@ void GatherForPredefinedReport(EnergyPlusData &state)
-                    PreDefTableEntry(state, state.dataOutRptPredefined->pdchFenParent, surfName, Surface(iSurf).BaseSurfName);
-                    curAzimuth = Surface(iSurf).Azimuth;
-                    // Round to two decimals, like the display in tables
-                    curAzimuth = round(curAzimuth * 100.0) / 100.0;
-                    PreDefTableEntry(state, state.dataOutRptPredefined->pdchFenAzimuth, surfName, curAzimuth);
-                    isNorth = false;
-                    curTilt = Surface(iSurf).Tilt;
-                    PreDefTableEntry(state, state.dataOutRptPredefined->pdchFenTilt, surfName, curTilt);
-                    if ((curTilt >= 60.0) && (curTilt < 180.0)) {
-                        if ((curAzimuth >= 315.0) || (curAzimuth < 45.0)) {
-                            PreDefTableEntry(state, state.dataOutRptPredefined->pdchFenDir, surfName, "N");
-                            isNorth = true;
-                        } else if ((curAzimuth >= 45.0) && (curAzimuth < 135.0)) {
-                            PreDefTableEntry(state, state.dataOutRptPredefined->pdchFenDir, surfName, "E");
-                        } else if ((curAzimuth >= 135.0) && (curAzimuth < 225.0)) {
-                            PreDefTableEntry(state, state.dataOutRptPredefined->pdchFenDir, surfName, "S");
-                        } else if ((curAzimuth >= 225.0) && (curAzimuth < 315.0)) {
-                            PreDefTableEntry(state, state.dataOutRptPredefined->pdchFenDir, surfName, "W");
-                        }
+                }
+                curWSC = Surface(iSurf).activeWindowShadingControl;
+                // compute totals for area weighted averages
+                fenTotArea += windowAreaWMult;
+                ufactArea += nomUfact * windowAreaWMult;
+                shgcArea += SHGCSummer * windowAreaWMult;
+                vistranArea += TransVisNorm * windowAreaWMult;
+                if (isNorth) {
+                    fenTotAreaNorth += windowAreaWMult;
+                    ufactAreaNorth += nomUfact * windowAreaWMult;
+                    shgcAreaNorth += SHGCSummer * windowAreaWMult;
+                    vistranAreaNorth += TransVisNorm * windowAreaWMult;
+                } else {
+                    fenTotAreaNonNorth += windowAreaWMult;
+                    ufactAreaNonNorth += nomUfact * windowAreaWMult;
+                    shgcAreaNonNorth += SHGCSummer * windowAreaWMult;
+                    vistranAreaNonNorth += TransVisNorm * windowAreaWMult;
+                }
+                // shading
+                if (Surface(iSurf).HasShadeControl) {
+                    PreDefTableEntry(state, state.dataOutRptPredefined->pdchFenSwitchable, surfName, "Yes");
+                    PreDefTableEntry(state, state.dataOutRptPredefined->pdchWscName, surfName, state.dataSurface->WindowShadingControl(curWSC).Name);
+                    // shading report
+                    std::vector<std::string> WindowShadingTypeStr = {
```

Comment:
`constexpr std::array<std::string_view>` and same for the next one. May want to move to the .hh file also if they are replicated elsewhere.